### PR TITLE
feat(helm): add support for additional `PodLogs` pipeline stages in `selfMonitoring`

### PR DESCRIPTION
**Pull request by dannykopping**
**What this PR does / why we need it**:
The `PodLogs` resource currently only supports a single pipeline stage (`cri`), and I need it to at least support configuring the `docker` stage as well; figured it'd make more sense to just allow arbitrary stages.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:
Miss you guys!!

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: production/helm/loki/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,11 @@ Entries should include a reference to the pull request that introduced the chang
 
 [//]: # (<AUTOMATED_UPDATES_LOCATOR> : do not remove this line. This locator is used by the CI pipeline to automatically create a changelog entry for each new Loki release. Add other chart versions and respective changelog entries bellow this line.)
 
+## 5.47.2
+
+- [ENHANCEMENT] Allow for additional pipeline stages to be configured on the `selfMonitoring` `Podlogs` resource.
```
Comment (MichelHollands) [inline production/helm/loki/CHANGELOG.md:18]: This is duplicate

**Review by MichelHollands — APPROVED** [review]
Hey Danny, how are things? I left 1 comment, otherwise looking fine.

**Comment by dannykopping** [comment]
> Hey Danny, how are things? I left 1 comment, otherwise looking fine.

All good man! Hope all's well with you too?
Thanks for the quick review.

I'll likely be pushing a few more enhancements in the next few days & weeks; we're going to be rolling the Loki SSD chart in a separate observability chart for https://github.com/coder/coder ❤️

**Comment by dannykopping** [comment]
@MichelHollands could you approve the build please?

**Comment by MichelHollands** [comment]
> @MichelHollands could you approve the build please?

Sorry, I forgot that we have to do that for your PRs now. It's running.

**Comment by slim-bean** [comment]
@dannykopping just a heads up that we are deprecating the monitoring sections of this chart in the next release (very soon with 3.0) and encouraging folks to use https://github.com/grafana/meta-monitoring-chart

**Pull request merged**